python/lsst/ts/scheduler/scheduler_csc.py

```diff
@@ -180,7 +221,8 @@ def do_exitControl(self, id_data):
         id_data : `salobj.CommandIdData`
             Command ID and data
         """
-        self._do_change_state(id_data, "exitControl", [base_csc.State.STANDBY], base_csc.State.OFFLINE)
+        self._do_change_state(id_data, "exitControl", [base_csc.State.FAULT,
+                                                       base_csc.State.STANDBY], base_csc.State.OFFLINE)
```

Comment:
So, the `enterControl` command is the one that is not implemented in the BaseCSC by default. The Scheduler is one of those CSCs that starts in OFFLINE and then need to transition to STANDBY. Also, when we go from STANDBY to OFFLINE the scheduler will not exit, but rather re-read and publish the available settings and wait to be put in standby. We may also need to perform some operations while in offline. 

I will implement your suggestions on enable, disable and start.

Reply:
Ouch. I suppose you are sure we need that capability (as opposed doing these "offline" operations in STANDBY mode instead)? I'm afraid it's going to be a bit ugly to subclass BaseCsc to make an externally commandable component, but I hope it works.

Reply:
While I was changing the `do_` methods to the `beg_` I realized why I did it that way. Basically, I need to have those tasks as coroutines because they actually take some time to run. If I use `bed_` or `end_` they would have to be regular methods and I could compromise the responsiveness of the CSC.

Reply:
Ah. That makes sense. I now wish I'd made it all coroutines. Maybe for version 4...